utils/queuer.go

```diff
@@ -17,6 +18,8 @@ type KVQueue interface {
 	Enqueue(key Key, value codec.ProtoMarshaler)
 	Dequeue(value codec.ProtoMarshaler, filter ...func(value codec.ProtoMarshaler) bool) bool
 	IsEmpty() bool
+	ExportState() (state QueueState)
+	ImportState(state QueueState, validator ...func(QueueState) error)
```

Comment:
I'd rather have this in a separate interface (if it's needed at all)
To me, exporting and importing state is not essential to the behaviour of a queue 😄

Reply:
I'll just remove them from the interface.